src/CMakeLists.txt

```diff
@@ -75,31 +75,37 @@ elseif(CMAKE_SYSTEM_NAME STREQUAL "Windows")
 endif()
 unset(${PROJECT_NAME}_soversion)
 
+add_library(secp256k1_warning_suppression INTERFACE)
+target_compile_options(secp256k1_warning_suppression INTERFACE
+  # Disable warning C4996 for the usage of a function, class member, variable, or typedef that's marked deprecated.
+  $<$<C_COMPILER_ID:MSVC>:/wd4996>
+)
```

Comment:
Why treat this special and not add it using `try_append_c_flags`?

Reply:
Isn't it better to catch such warnings in non-test/bench code?

Reply:
I'm not convinced that this justifies making things more complicated in the build system.

Hm, I just checked. The deprecated warnings here are all the same kind:

```
src/tests.c(7695,27): warning C4996: 'getenv': This function or variable may be unsafe. Consider using _dupenv_s instead. To disable deprecation, use _CRT_SECURE_NO_WARNINGS. See online help for details.
```

And we really don't want *these* warnings. We can't use `_dupenv_s` because it exists only on Windows. We can't use `getenv_s` because it requires C11 Annex K. But can we just pass `/D_CRT_SECURE_NO_WARNINGS` instead of `/wd4996`?  That's narrower, and then I think there's really no need to restrict it to test/bench/etc.

Reply:
Thanks! [Updated](https://github.com/bitcoin-core/secp256k1/pull/1328#issuecomment-1562534860).